### PR TITLE
Feature flags

### PR DESCRIPTION
**Pull request by koke**
Instead of having a bunch of checks for `DEBUG` and `INTERNAL_BUILD` spread across the app, I suggest we keep them contained and actually check for enabled features.

This moves the checks from compile time to run time, so it might not be best for all scenarios:
- Constants that have different values for each build
- Static tables with row constants that depend on features, but we're moving those to ImmuTable progressively
- Code that we don't just want disabled but not linked (e.g. HockeySDK)

Despite all this, I think this approach is interesting so we can easily start using more feature flags. Ideally we'd start work on a new feature by flagging it and enabling only on Debug builds. This way, we can make smaller PRs and only release the feature when it's ready.

I've only added the FeatureFlag code (+tests), I'll start using it on a separate PR

Needs Review: @SergioEstevao

### CODE REVIEW

File: WordPress/Classes/Utility/FeatureFlag.swift [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    /// Production build released in the app store
+    case AppStore
+
+    /// Returns the current build type
+    static var current: Build {
+        if let override = _overrideCurrent {
+            return override
+        }
+
+        #if DEBUG
```
Comment (SergioEstevao) [inline WordPress/Classes/Utility/FeatureFlag.swift]: On the PR description you say that the checks pass from compile to run time,  but this looks to be set on compile time?

**Comment by SergioEstevao** [comment]
@koke I like the PR, but are we sure we want to be limited to the build type only? I know that in some parts of the app we where previous using a special URL scheme to activate features. Do you plan to support this?

File: WordPress/Classes/Utility/FeatureFlag.swift (hunk 1 above)
Comment (koke) [inline WordPress/Classes/Utility/FeatureFlag.swift]: I think that bit wasn't clear. The build type is still defined by build parameters, but using this code does the check at runtime. For instance:

```swift
#if DEBUG
  doStuff() // this isn't included in the final binary
#endif
```

```swift
if build(.Debug) {
  doStuff() // this is included in the binary, but it's never reached
}
```

**Comment by koke** [comment]
> are we sure we want to be limited to the build type only?

It's not limited. In `FeatureFlag.enabled` you can build your rules however you want. My first goal was to disable some stuff on App Store builds, so that's why the examples show that. But I'd also like to have some way to tweak those in the future, maybe even use flags that we get from the API for a specific wp.com account. I think this would still work as the frontend to all those checks.

**Comment by SergioEstevao** [comment]
:shipit:

File: WordPress/WordPressTest/FeatureFlagTest.swift [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import Nimble
+import XCTest
+@testable import WordPress
+
+class FeatureFlagTest: XCTestCase {
+
+    func testBuild() {
```
Comment (sendhil) [inline WordPress/WordPressTest/FeatureFlagTest.swift]: Pretty looking tests ;)

**Pull request merged**